examples/ex_adaptive.ipynb

```diff
@@ -0,0 +1,503 @@
+    "    bisection_mode_condition=None, # None (the default) here gives only logarithmic bisection when adding internal points\n",
+    "    max_iterations=10000,\n",
+    ")\n",
+    "\n",
+    "transform_arr_short = nft.fourier_integral_fixed_sampling(\n",
+    "    times=times,\n",
+    "    frequencies=frequencies_short,\n",
+    "    func_values=func_arr_short,\n",
+    "    pos_inf_correction_term=True,\n",
+    "    neg_inf_correction_term=False,\n",
+    "    interpolation=\"pchip\"\n",
+    ")\n",
+    "\n",
+    "print(f\"Final number of frequencies: {len(frequencies_short)}, spanning from {frequencies_short[0]} to {frequencies_short[-1]}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Select component to plot, feel free to change np.real to e.g. np.imag, np.abs, or np.angle\n",
+    "f1 = np.real(transform_arr_short)\n",
+    "f2 = np.real(transform_arr_analytic)\n",
+    "\n",
+    "fig, (ax1, ax2, ax3) = plt.subplots(1,3,figsize=(14,5))\n",
+    "ax1.plot(times, f1, \"-o\", markersize=4, label=\"neffint adaptive\")\n",
+    "ax1.plot(times, f2, \"-o\", markersize=4, label=\"analytic\")\n",
+    "ax1.semilogx() # Feel free to change to ax1.loglog\n",
+    "ax1.legend()\n",
+    "ax1.set_title(\"Fourier transform outputs\")\n",
+    "ax1.set_xlabel(\"t [a.u.]\")\n",
+    "\n",
+    "ax3.hlines(1e0, times[0], times[-1], \"r\", linestyles=\"dotted\", label=\"Tolerance\")\n",
+    "for ax, diff in [(ax2, relative_diff), (ax3, absolute_diff)]:\n",
+    "    ax.plot(times, diff(f1, f3), \"-o\", markersize=4, label=\"adaptive vs analytic\")\n",
+    "    ax.loglog()\n",
+    "    ax.legend()\n",
+    "    ax.set_title(diff.__doc__)\n",
+    "    ax.set_xlabel(\"t [a.u.]\")\n",
+    "\n",
+    "plt.tight_layout()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, we see that the results are much worse than both previous rounds. The absolute error is even above the tolerance of $10^0$ for the entire range! It is therefore advisable to experiment to find a suitable range of initial frequencies. Remember that both $0$ and $\\pm \\infty$ are valid entries into the initial frequency range. The only things to note is that `func` should give finite outputs for all frequencies in the initial range, and that, as mentioned above, adding infinities can end up hurting accuracy instead of helping it."
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "neffint",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.7.12"
+  },
+  "orig_nbformat": 4
+ },
+ "nbformat": 4,
+ "nbformat_minor": 2
+}
```

Comment:
this example is really excellent! it is very didactic, clear and detailed. Thanks a lot!